### DISCUSSION TITLE
I like Python, but why not build Mojo on top of Swift?

### DISCUSSION THREAD

**Discussion by dleuck** [modular/modular#485] [General]
@lattner Python has aged better than any other language from its time, but Swift, a brilliant gift from you, obviously benefits from many advantages including the most robust type system in existence and type inferencing so good that it feels like a scripting language. Its elegance is unmatched. I read about the reasons you moved on, but couldn't Mojo be an AI-oriented evolution of Swift or at least take inspiration from it?

**Comment by mojodojodev** [comment]
The main reason is to leverage the existing Python ecosystem that is dominating AI. There are a lot of lessons and concepts being bought over from Swift which was optimized to bridge the existing Apple Objective-C ecosystem. Mojo is optimized to bridge the existing Python ecosystem so all those AI researchers and data scientists can continue using the syntax and libraries they love, and systems programmers can build high performance libraries for them, by taking advantage of all the great AI compiler technology that has been getting built with MLIR. Swift for Tensorflow didn't get much adoption because AI researchers aren't interested in learning a new language, they just want to keep using Python. The big advantage of Mojo is that it doesn't have to take users away from Python, they compliment each other so as one becomes more popular the other will as well.

In a lot of ways it is an evolution of Swift and does take a lot of inspiration from it in terms of implementation details, it's just the trivial things like braces and syntax where it's more similar to Python.

**Comment by lattner** [comment]
Hi @dleuck, I'm also a fan of Swift.  The major issue with it is that it isn't a member of the Python family.  I'm confused why you think that Mojo isn't inspired by Swift, because Mojo certainly is.  That said, you're right that building a new thing takes longer than leveraging an existing thing.  The reason we're taking this approach is that we're optimizing for quality of result (we want to build the world's best thing) not time to market.  I would consider it success if you come to love Mojo more than Swift some day, :)

-Chris
